docs/rfc/rfc-1.adoc

```diff
@@ -225,9 +234,17 @@ conditions for the btc fee and then off-chain see which deposits *would be*
 swept if we *were* to sweep. If the sum of those deposits exceeds the
 `sweep_max_btc` parameter, then we should initiate a sweep.
 
-After submitting the transaction, it will cost some amount of bitcoin based on
-what miners are charging for the bitcoin fee. This fee comes out of the unswept
-deposits. When the transaction clears, and the information has made its way
+The sweeping transaction will cost some amount of bitcoin based on what miners
+are charging for the bitcoin fee in the current market conditions. The fee is
+paid proportionally from the unswept deposits. Once the transaction is
+submitted to the bitcoin mempool, the miners will either include it in a block
+within `sweeping_fee_bump_period` or not. If they don't, then we increment a
+fee multiplier: `fee_multiplier = fee_multiplier +
+sweeping_fee_multiplier_increment` and then calculate the new fee: `fee =
+base_fee * fee_multiplier`. We repeat until either the transaction posts or
+`sweeping_fee_multiplier_increment` exceeds `sweeping_fee_max_multiplier`.
```

Comment:
Well, I think that there may be situations when we should stop multiplication earlier, because wee need to also take under consideration what was the `max_fee` declared by the owners of the deposits.

This relates to the Q20 which I asked on Discord:
> 20. Will we be able to bump up the fee on the sweeping tx only until we hit the `max_fee` of one of the depositors selected to the sweep?
Let’s say there were four revealed deposits:
utxo_1, max_fee: 100 satoshi
utxo_2, max_fee: 50 satoshi
utxo_3, max_fee: 21 satoshi
utxo_4, max_fee: 10 satoshi
Let’s say algorithm determined that utxos 1-3 should be included in the sweep operation and that 60 satoshi should be a starting fee for the sweeping operation (that gives 20 satoshi per deposit, which is lower that `max_fee` of each depositor).
Now let’s assume that the tx wasn’t mined and it was time to increase the offered fee by 10%, making the fee equal to 66 satoshi (22 satoshi per depositor). The new fee per depositor would be now higher than `max_fee` of the third deposit. What we do in such scenario?
A) Replace old sweeping tx (fee: 60) with a new one (fee: 66), using same utxos, and if it gets mined, charge depositors this way:
>    depositor 3: 21 satoshi
>    depositors 1 and 2: (66-21)/2 satoshi
> B) Replace old sweeping tx (fee: 60) with a new one, involving just utxo_1 and utxo_2 (let’s say that with just 2 deposits fee would be estimated to 50). // Not sure if we can do such replacement in one transaction...
C) Stop bumping up fee and wait patiently until the original tx gets mined
D) ... something I haven’t thought of?

To avoid this problem we could do the packing of the UTXOs more cautiously and choose only those UTXOs that leave us a margin for bumping of the fee to the max `fee_multiplier`. But this way less UTXOs would qualify for the sweep...

Reply:
> To avoid this problem we could do the packing of the UTXOs more cautiously and choose only those UTXOs that leave us a margin for bumping of the fee to the max fee_multiplier. But this way less UTXOs would qualify for the sweep...

the current algorithm assumes that all of the included UTXO's have a`max_fee` >= `sweeping_fee_max_multiplier * base_fee` otherwise you run into exactly what you're talking about above

also, maybe for some expectation setting

https://privacypros.io/tools/bitcoin-fee-estimator/

the current bitcoin fee for a normal segwit transaction is $0.73 !

Reply:
Oh - before I forget: I know the information is fragmented, but this is mentioned on [L388-390](https://github.com/keep-network/tbtc-v2/pull/65/files#diff-1044c578112615d9cb745f17e8203a444e72ddb3c5d8d5fd97d97038952c77b1R388-R390)

Reply:
Oh, right, I missed that.